Utilities/HttpServer.cs

```diff
@@ -78,9 +78,11 @@ public class HttpServer {
         return false;
 
       try {
-        listenerThread.Abort();
-        listener.Stop();
-        listenerThread = null;
+        if (listenerThread != null){

```

Comment:
Its a leagl state, that listenerThread is null.
In your sequence listener.Stop() would not be called.
It would be better to just handle the listenerThread inside the if-block.
```csharp
        if (listenerThread != null) {
          listenerThread.Abort();
          listenerThread = null;
        }
        listener.Stop();
```

Reply:
ipinfo:master

Reply:
> Its a leagl state, that listenerThread is null.
> 
> In your sequence listener.Stop() would not be called.
> 
> It would be better to just handle the listenerThread inside the if-block.
> 
> ```csharp
> 
>         if (listenerThread != null) {
> 
>           listenerThread.Abort();
> 
>           listenerThread = null;
> 
>         }
> 
>         listener.Stop();
> 
> ```